src/main/java/org/apache/commons/jxpath/functions/MethodFunction.java

```diff
@@ -34,6 +36,8 @@ public class MethodFunction implements Function {
     private final Method method;
     private static final Object[] EMPTY_ARRAY = {};
 
+    private JXPathFilter jxpathFilter = new SystemPropertyJXPathFilter();
```

Comment:
It would be probably better to add the same filtering mechanism to `ConstructorFunction`. I think it is not vulnerable now, but we might do this, just to be sure it will not become vulnerable in new versions.